.travis.yml

```diff
@@ -21,13 +17,8 @@ env:
 notifications:
   email: false
 
-addons:
-  apt:
-    packages:
-      - dvipng
-      - texlive-latex-base
-      - texlive-latex-extra
-
 install: source build_tools/travis/install.sh
-script: bash build_tools/travis/test_script.sh
-after_success: source build_tools/travis/after_success.sh
+
+script:
+  - docker build -t allennlp/allennlp .
+  - docker run allennlp/allennlp -c /bin/bash -c "/stage/build_tools/travis/test_script.sh"
```

Comment:
Building the image itself runs our tests, so this line seems redundant.

Why do you want everything to run from the docker image?  Why not just add a test condition that ensures that the docker image builds?  We could even just replace the current "run tests" one with one that builds the docker image, but only if it uploads coverage information also - the coverage CI is not something I want to get rid of.

Reply:
@matt-gardner I do not get rid of the coverage CI in this PR. 

I would like to run everything from the docker image so we have one fewer environment.  Presently, it's possible that a test would fail within Travis, but not within our Docker environment.

Reply:
@matt-gardner if we added a test condition that ensures the docker image builds, why wouldn't we then use that docker image to run the rest of our tests?

Reply:
The other justification is simply that this PR removes 50 lines of code.

Reply:
My main concern is that building the docker image is going to dramatically increase the time travis takes to run our tests.  If that's not true, and we still get the same parallelism and functionality that we have now with plain travis, I don't much care how it works.

Reply:
+1 for keeping coverage, generally with Matt on the idea that this seems fine as is and any optimisation shouldn't make it slower, but fine if its faster.

Reply:
Just to be clear--I never advocated removing coverage.

Reply:
And my comment about coverage was wondering if it would work when run inside of docker. The way I was imagining running the tests seemed like it would be hard.